tests/fsharp/typecheck/overloads/neg_System.Threading.Tasks.Task.Run.OverloadList.bsl

```diff
@@ -0,0 +1,48 @@
+
+neg_System.Threading.Tasks.Task.Run.OverloadList.fsx(5,11,5,35): typecheck error FS0041: A unique overload for method 'Run' could not be determined based on type information prior to this program point. A type annotation may be needed.
+
+Known type of argument: (unit -> Task<int>)
+
+Candidates:
+ - Task.Run<'TResult>(function: Func<'TResult>) : Task<'TResult>
+ - Task.Run<'TResult>(function: Func<Task<'TResult>>) : Task<'TResult>
+
+neg_System.Threading.Tasks.Task.Run.OverloadList.fsx(6,11,6,44): typecheck error FS0041: A unique overload for method 'Run' could not be determined based on type information prior to this program point. A type annotation may be needed.
+
+Known type of argument: Func<Task<int>>
+
+Candidates:
+ - Task.Run<'TResult>(function: Func<'TResult>) : Task<'TResult>
+ - Task.Run<'TResult>(function: Func<Task<'TResult>>) : Task<'TResult>
```

Comment:
Note that this error message highlights something where I find the compiler not trying hard enough, here `Func<Task<int>>` is theoritically matching `Func<Task<'TResult>>` overload, in the older compiler, it shows:

> A unique overload for method 'Run' could not be determined based on type information prior to this program point. A type annotation may be needed. Candidates: Task.Run<'TResult>(function: Func<'TResult>) : Task<'TResult>, Task.Run<'TResult>(function: Func<Task<'TResult>>) : Task<'TResult>

Where the user lacked the `Func<Task<int>>` being the known type of argument and had no type subsumption error either.

I feel the ` A type annotation may be needed.` message should turn into more specifically pointing at requesting the generic type argument in that case.

I have not investigated but felt it would be for, worst case, another round of enhancements on that message in separate PR, or best case, adding a step to overload resolution to pick that overload.

@gusty mentionned to me having the spec of current behaviour rewritten more formally, then adding such cases would be the approach.

Reply:
Yes, makes sense as an area for subsequent improvement after this PR